src/DependencyInjection/SonataMediaExtension.php

```diff
@@ -455,12 +455,18 @@ public function configureFilesystemAdapter(ContainerBuilder $container, array $c
             ;
 
             if (3 === $config['filesystem']['s3']['sdk_version']) {
-                $container->getDefinition('sonata.media.adapter.service.s3')
```

Comment:
What about writing something like this:

```
$arguments = [
    'credentials' => false,
    'region' => $config['filesystem']['s3']['region'],
    'version' => $config['filesystem']['s3']['version'],
];

if (isset($config['filesystem']['s3']['secretKey'], $config['filesystem']['s3']['accessKey'])) {
    $arguments['credentials'] = [
        'secret' => $config['filesystem']['s3']['secretKey'],
        'key' => $config['filesystem']['s3']['accessKey'],
    ];
}

$container->getDefinition('sonata.media.adapter.service.s3')
    ->replaceArgument(0, $arguments);
```
?

Reply:
You're right, it's more readable